pommerman/network/ion_client/network.py

```diff
@@ -0,0 +1,186 @@
+                    "intent": constants.NetworkCommands.check.value
+                })
+            )
+            message_recieved = rapidjson.loads(self.ws_.recv())
+            self.lock.release()
+            if message_recieved["intent"
+                               ] == constants.NetworkCommands.status_ok.value:
+                return [
+                    message_recieved["players"], message_recieved["matches"]
+                ]
+            elif message_recieved[
+                "intent"
+            ] == constants.NetworkCommands.status_full.value:
+                raise Exception(constants.Exceptions.net_server_full.value)
+            raise Exception(constants.Exceptions.net_respond_fail.value)
+        except:
+            self.lock.release()
+            raise Exception(constants.Exceptions.net_connect_fail.value)
+
+    def join_list(self, room=False):
+        """Argument:  
+        * room: The room to be created/joined. If False, the public room will \
+be joined
+        Description: Check if the server actually responds"""
+        self.lock.acquire()
+        if not room:
+            self.ws_.send(
+                rapidjson.dumps(
+                    {
+                        "intent": constants.NetworkCommands.match.value
+                    }
+                )
+            )
+        else:
+            self.ws_.send(
+                rapidjson.dumps(
+                    {
+                        "intent": constants.NetworkCommands.room.value,
+                        "room": room
+                    }
+                )
+            )
+        self.lock.release()
+        decoded_msg = rapidjson.loads(self.ws_.recv())
+        if (
+            decoded_msg["intent"] == constants.NetworkCommands.status_full.value
+        ):
+            if room:
+                raise Exception(constants.Exceptions.room_full.value)
+            else:
+                raise Exception(constants.Exceptions.match_full.value)
+        self.id = decoded_msg["player_id"]
+        self.mode = decoded_msg["mode"]
+
+    def waitm(self):
+        """Description: Wait for a response from the server regarding a match 
+        request"""
+        self.lock.acquire()
+        try:
+            r = rapidjson.loads(self.ws_.recv())
+            if r["intent"] == constants.NetworkCommands.match_start.value:
+                self.match_id = r["match_id"]
+        except:
+            self.lock.release()
+            raise Exception(constants.Exceptions.net_server_closed.value)
+        self.lock.release()
+
+    def match_get(self):
+        """Description: Get the next step of the match"""
+        self.lock.acquire()
+        try:
+            message_recieved = self.ws_.recv()
+        except:
+            self.lock.release()
+            raise Exception(constants.Exceptions.net_respond_fail.value)
+        self.lock.release()
+        try:
+            r = rapidjson.loads(str(gzip.decompress(message_recieved), "utf-8"))
```

Comment:
what is `r` supposed to be? write out the name.

Reply:
Also what is going on here? Why do you expect this to fail?

Reply:
I elaborated on that in a comment I added. `# Messages with normal match data are compressed using GZIP while match end notifications aren't. So we move on to that if this fails to decompress with gzip.`